S7.Net/Protocol/Serialization.cs

```diff
@@ -8,39 +8,43 @@ internal static class Serialization
     {
         public static ushort GetWordAt(IList<byte> buf, int index)
         {
-            return (ushort) ((buf[index] << 8) + buf[index]);
+            return (ushort)((buf[index] << 8) + buf[index]);
         }
 
         public static byte[] SerializeValue(object value)
         {
             switch (value.GetType().Name)
             {
                 case "Boolean":
-                    return new[] {(byte) ((bool) value ? 1 : 0)};
+                    return new[] { (byte)((bool)value ? 1 : 0) };
                 case "Byte":
-                    return Types.Byte.ToByteArray((byte) value);
+                    return Types.Byte.ToByteArray((byte)value);
                 case "Int16":
-                    return Types.Int.ToByteArray((Int16) value);
+                    return Types.Int.ToByteArray((Int16)value);
                 case "UInt16":
-                    return Types.Word.ToByteArray((UInt16) value);
+                    return Types.Word.ToByteArray((UInt16)value);
                 case "Int32":
-                    return Types.DInt.ToByteArray((Int32) value);
+                    return Types.DInt.ToByteArray((Int32)value);
                 case "UInt32":
-                    return Types.DWord.ToByteArray((UInt32) value);
+                    return Types.DWord.ToByteArray((UInt32)value);
                 case "Double":
-                    return Types.Double.ToByteArray((double) value);
+                    return Types.Double.ToByteArray((double)value);
+                case "Single":
+                    return Types.Single.ToByteArray((float)value);
                 case "Byte[]":
-                    return (byte[]) value;
+                    return (byte[])value;
                 case "Int16[]":
-                    return Types.Int.ToByteArray((Int16[]) value);
+                    return Types.Int.ToByteArray((Int16[])value);
                 case "UInt16[]":
-                    return Types.Word.ToByteArray((UInt16[]) value);
+                    return Types.Word.ToByteArray((UInt16[])value);
                 case "Int32[]":
-                    return Types.DInt.ToByteArray((Int32[]) value);
+                    return Types.DInt.ToByteArray((Int32[])value);
                 case "UInt32[]":
-                    return Types.DWord.ToByteArray((UInt32[]) value);
+                    return Types.DWord.ToByteArray((UInt32[])value);
                 case "Double[]":
-                    return Types.Double.ToByteArray((double[]) value);
+                    return Types.Double.ToByteArray((double[])value);
+                case "Single[]":
+                    return Types.Single.ToByteArray((float[])value);
```

Comment:
I'm wondering what you're using for the formatting, because I guess the default R# settings will actually have a space between type and value in a cast.

Reply:
Nope, my VS doesn't do that. Spacing after cast is disabled and never touched it (so is [default setting](https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference#spacing)). Look at TSDU class, there aren't spaces after cast either, so i guess my setting is commonly used.